[CAS-1072] Reorganize client docs (#1977)
GetStream/stream-chat-android: docusaurus/docs/Android/02-client/04-events.md

# Events

> Events allow the client to stay up to date with changes to the chat. Examples are a new message, a user's image that updated, a reaction, or a member joining the channel.

Reviewer: ```suggestion
Events allow the client to stay up to date with changes to the chat. For example, you get events when a new message was sent, a user's image was updated, a reaction was added, or a member joined the channel.
```

> Events are only received when SDK is connected to the socket and the application is in the foreground.

Reviewer: ```suggestion
Events are only received when the client is connected to the socket and the application is in the foreground.
```

Plus we could also link to push notifications here, as that's what you get if the socket connection is not active
